### DISCUSSION TITLE
Route duration from profile / speed

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#28] [Q&A] [answered]
We can have 2 types for the route duration:
- Route duration provided from routing services
- Route duration calculated from selected navigation profile + average speed (https://github.com/devemux86/cruiser/discussions/19)

Note that each routing service calculates differently the duration of routes (using road speeds, turn costs, etc.).

Better we have a setting to select the duration type and display / use it everywhere, UI, navigation, ...?

**Comment by DolomitesTom** [comment]
I tend more to the general setting to switch the values. The value in brackets could maybe be confusing, but is also something that you get used to. It is a great idea to have the route or wp distances calculated from the average speed... and also have the real average speed somewhere in statistics 😁 (already mentioned somewhere else).

**Comment by devemux86** [comment]
With the introduction of **navigation profiles** where you can freely select your own **average speed** (details in https://github.com/devemux86/cruiser/discussions/19),
now it is time for a new option to select whether the **duration of the route** and the times in the **voice navigation**:
- Will be provided by the routing services or imported routes
- Or calculated from your customized **navigation profile and speed**

The route duration setting will be available in the next Cruiser update.

**Reply by DolomitesTom** [reply]
perfect, I am already looking forward to it.

**Comment by devemux86** [comment, marked as answer]
[Cruiser 3.0.9](https://play.google.com/store/apps/details?id=gr.talent.cruiser) is available with route duration option from:
- Route (calculated or imported)
- Speed (profile / average speed)

This affects the calculated and imported routes and the times in _navigation_.

- Settings | Routing | Options | Route duration
- Or search settings for "duration"
